src/python/grpcio_tests/tests_aio/unit/call_test.py

```diff
@@ -109,6 +109,24 @@ class TestUnaryUnaryCall(AioTestBase):
             call = hi(messages_pb2.SimpleRequest())
             self.assertEqual('', await call.details())
 
+    async def test_call_initial_metadata_awaitable(self):
+        async with aio.insecure_channel(self._server_target) as channel:
+            hi = channel.unary_unary(
+                '/grpc.testing.TestService/UnaryCall',
+                request_serializer=messages_pb2.SimpleRequest.SerializeToString,
+                response_deserializer=messages_pb2.SimpleResponse.FromString)
+            call = hi(messages_pb2.SimpleRequest())
+            self.assertEqual((), await call.initial_metadata())
```

Comment:
Can we add one more test for non-empty initial_metadata and trailing_metadata? Also we might need tests for invalid metadata, see `_invalid_metadata_test.py`.

Optional: I felt like we could create a separate file for metadata, since their content are kind of isolated.

Reply:
I noticed the sync version is in a seperate file.